### PR TITLE
[CDAP-12836] Adds one time to copy data from S3 and GCS

### PR DESCRIPTION
**Pull request by ajainarayanan**
JIRA: https://issues.cask.co/browse/CDAP-12836

- One time copy from S3 to CDAP Dataset works fine.
- WIP for GCS to CDAP dataset.

### CODE REVIEW

File: cdap-ui/app/cdap/components/DataPrep/ColumnActionsDropdown/index.js [hunk 1]
```diff
@@ -220,6 +219,7 @@ export default class ColumnActionsDropdown extends Component {
   }
 
   componentWillUnmount() {
+    this.eventEmitter.off('CLOSE_POPOVER', this.toggleDropdown.bind(this, false));
```
Comment (elfenheart) [inline cdap-ui/app/cdap/components/DataPrep/ColumnActionsDropdown/index.js:222]: what's the reason for this change? this can cause multiple dropdown to open, right?

Comment (ajainarayanan) [reply cdap-ui/app/cdap/components/DataPrep/ColumnActionsDropdown/index.js:222]: Actually its not. The `CLOSE_POPOVER` event is emitted only when we `execute` any directive. The `this.documentClick$` in line 249 closes the popover if the user clicks anywhere else in the document. I am not sure why we were listening to this event in the `constructor` and switching it off in `componentWillMount`. But moving it to `componentWillUnmount` didn't change and the popover closes as it used to be. 

There is an inherent JS error that is happening with popover that I was trying to debug. Couldn't get to it in time. Will see if I can fix the js error.

**Review by elfenheart — APPROVED** [review]
👍  LGTM

File: cdap-ui/app/cdap/components/DataPrep/TopPanel/AddToPipelineModal.js [hunk 2]
```diff
@@ -206,7 +219,10 @@ export default class AddToHydratorModal extends Component {
     // This is a hack.. should not do this
     // We are still shipping kafka-plugins with hydrator-plugins 1.7 but
     // it doesn't contain the streamingsource or batchsource plugins
-    let pluginArtifact = artifactsList.filter((artifact) => artifact.name === 'kafka-plugins');
+    let pluginArtifact = find(artifactsList, {name: 'kafka-plugins'});
+    if (!pluginArtifact) {
+      return T.translate(`${PREFIX}.kafkaPipeline.pipelineError`);
+    }
     pluginArtifact = pluginArtifact[pluginArtifact.length - 1];
```
Comment (tonybach) [inline cdap-ui/app/cdap/components/DataPrep/TopPanel/AddToPipelineModal.js]: If you're using `find` now then you don't need this line anymore right, since it already only returns the one element?

File: cdap-ui/app/cdap/components/DataPrep/TopPanel/PipelineConfigHelper.js [hunk 3]
```diff
@@ -169,7 +179,10 @@ function constructKafkaSource(artifactsList, kafkaInfo) {
   // This is a hack.. should not do this
   // We are still shipping kafka-plugins with hydrator-plugins 1.7 but
   // it doesn't contain the streamingsource or batchsource plugins
-  let pluginArtifact = artifactsList.filter((artifact) => artifact.name === 'kafka-plugins');
+  let pluginArtifact = find(artifactsList, {name: 'kafka-plugins'});
+  if (!pluginArtifact) {
+    return T.translate('features.DataPrep.TopPanel.kafkaPipeline.ingestDataError');
+  }
   pluginArtifact = pluginArtifact[pluginArtifact.length - 1];
```
Comment (tonybach) [inline cdap-ui/app/cdap/components/DataPrep/TopPanel/PipelineConfigHelper.js]: same here

**Comment by tonybach** [comment]
Code LGTM 👍

**Comment by bdmogal** [comment]
lgtm
